### PR TITLE
Procedure list

### PR DESCRIPTION
**Pull request by uxmal**
I've been annoyed by the limitations of the tree browser when displaying procedures. Among my complaints:
1. There is no way of selecting multiple items
1. There is no way of searching for procedures by name/address.
1. There is no way of sorting the procedures by name/address

I put this simple procedure list view together in a couple of hours:
image

Please let me know what you think and add your comments / questions to this PR.

### CODE REVIEW

File: src/Gui/IProcedureListService.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+using System.Threading.Tasks;
+
+namespace Reko.Gui
+{
+    public interface IProcedureListService : ICommandTarget
+    {
+        /// <summary>
+        /// The currently selected object in the project browser tree.
+        /// </summary>
+        Procedure SelectedProcedure { get; set; }
```
Comment (ptomin) [inline src/Gui/IProcedureListService.cs]: Unused property?

Comment (uxmal) [reply src/Gui/IProcedureListService.cs]: Not implemented yet, most likely :) Upcoming work will make it more "real".

**Comment by uxmal** [comment]
@Pavel: have you tried building this branch and testing it? Does it feel OK?

**Comment by ptomin** [comment]
I have built it. Looks OK.
One more thing. It would be nice to add possibility to show "Edit Signature" dialog from "Procedures" tab for selected procedure

**Comment by smx-smx** [comment]
Bug: When Using "Finish Decompilation" immediately after loading a binary, the list of procedures is empty

**Comment by uxmal** [comment]
Does this feel "cooked" enough to go to master?

**Pull request merged**